### DISCUSSION TITLE
GitOps Patterns team 🙂 Who wants to collaborate?

### DISCUSSION THREAD

**Discussion by scottrigby** [open-gitops/project#177] [Ideas]
## Update 2024

Please comment if you want to collaborate on compiling GitOps Patterns.

The need for official GitOps patterns has been discussed many times, but has been [bikeshedded](https://en.wikipedia.org/wiki/Law_of_triviality) because we couldn't agree what to call it (there were some good suggestions! But ultimately each had nits that prevented us from moving forward). We've decided what to call it, we just need to create the team and begin collaborating on compiling the content.

For history, see plenty of mentions in the [Historic: GitOps Working Group meetings doc (Dec 2020 – March 2024)](https://docs.google.com/document/d/1OrslXDEDOxnqiGon2JEhPAWfPt0DtlG2MOBidkoCDqc/edit) (search for "patterns" and "blueprints").

## Tasks

- [x] https://github.com/open-gitops/project/pull/197 PR in TEAMS.md
- [ ] Begin collaboration with the team (schedule a repeating day/time?)

## Original discussion topic

Per OpenGitOps meeting today, we decided to put a stronger effort into writing up GitOps patterns to help the community, because people definitely keep asking for this.

**Comment by christianh814** [comment]
+1 for this. Lots of great content by us individually that we can incorporate so we don't have to start from 0

**Comment by dtzar** [comment]
I am definitely interested to collaborate on this

**Comment by PrateekKumar1709** [comment]
I'm interested in contributing to this and would love to collaborate

**Comment by priyanka-ravi** [comment]
I’d love to help as well!

**Comment by schnatterer** [comment]
I've invested quite some time collection references to patterns and identifying them myself for [articles](https://cloudogu.com/en/blog/gitops-repository-patterns-part-1-introduction) and a [book](https://gitops-book.dev/) (published in German only, for now).
You can find the gist of it https://github.com/cloudogu/gitops-patterns.
I'd be honored if my work could be of use and am also open for contribution to this effort here, though I don't have a lot of time on my hands due to family and work 😇

**Comment by mbaykara** [comment]
+1 I'd love to contribute too!

**Comment by nishadmehendale** [comment]
+1

**Comment by carolili** [comment]
Hi, I use GitOps tools daily (not an expert though 😅) and I like documenting and writing. If that could be of value, I'd be happy to contribute!

**Comment by cwstrommer** [comment]
Interested in participating, specially when it comes to versioning and artifact promotion patterns.

**Comment by alonsomoya** [comment]
I'm in ☝

**Comment by dtzar** [comment]
One of the topics I'd like to try and establish is a standardized code/folder structure that handles the "80% use case".   I believe this will help people not be as overwhelmed to all the choices and opinions and tooling could be evolved to make the E2E scenarios easier.

Relevant content:
- https://github.com/christianh814/example-kubernetes-go-repo
- https://github.com/gnunn-gitops/standards/blob/master/folders.md
- https://fluxcd.io/flux/guides/repository-structure/
- https://developers.redhat.com/articles/2022/09/07/how-set-your-gitops-directory-structure
- https://www.youtube.com/watch?v=vLNZA_2Na_s
- https://github.com/gitops-bridge-dev
- https://github.com/Azure-Samples/aks-platform-engineering
- https://github.com/microsoft/multicluster-gitops

**Comment by aholbreich** [comment]
I'm in if still possible

**Comment by jonathanagustin** [comment]
Unsure if this is still open, but I would love to participate!

**Comment by blueskyjunkie** [comment]
I'm interested.
